crates/telio-traversal/src/endpoint_providers/local.rs

```diff
@@ -475,7 +447,8 @@ mod tests {
 
         let state = prepare_state_test(wg_mock, get_if_addrs_mock).await;
 
-        let interfaces = state.0.gather_local_interfaces().unwrap();
+        let interfaces = gather_local_interfaces(&state.0.get_if_addr).unwrap();
```

Comment:
I took a look for a little bit more at it and decided to propose more changes in regards to this functionality and ended up testing if it would work. In the end I've created a branch with the draft: #595 I think its cleaner and removes the need for generics. You can take a look, I think it's worth taking into you PR. Also it solves the issue I didn't like in your solution where you had a testing data for firewall.rs hardcoded into local_interfaces.rs

Reply:
I looked at your draft PR. I will incorporate that.